version: '3.9'
services:
  web:
    container_name: django_app
    build:
      context: .
      dockerfile: Dockerfile-web
    command: python manage.py runserver 0.0.0.0:8000
    volumes:
      - ./app/:/usr/src/app/
    ports:
      - "8000:8000"
    env_file:
      - ./.env.dev
    depends_on:
      - db
  db:
    container_name: db_container
    build:
      context: .
      dockerfile: Dockerfile-db
    volumes:
      - ./db_init:/docker-entrypoint-initdb.d/
      - postgres_data:/var/lib/postgresql/data/
    environment:
      POSTGRES_USER: "postgres"
      POSTGRES_PASSWORD: "postgres123"
      POSTGRES_DB: "PlacesRememberdb"
      PGDATA: "/var/lib/postgresql/data/pgdata"
    ports:
      - "5432:5432"
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres -d PlacesRememberdb" ]
      interval: 10s
      timeout: 5s
      retries: 5
      start_period: 10s
    restart: unless-stopped


  pgadmin:
    container_name: pgadmin_container
    image: dpage/pgadmin4:latest
    environment:
      PGADMIN_DEFAULT_EMAIL: "example@gmail.com"
      PGADMIN_DEFAULT_PASSWORD: "postgres123"
      PGADMIN_CONFIG_SERVER_MODE: "False"
    volumes:
      - pgadmin-data:/var/lib/pgadmin
    ports:
      - "5050:80"
    restart: unless-stopped

volumes:
  postgres_data:
  pgadmin-data: